tests/std/tests/P0896R4_ranges_alg_transform_binary/test.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+using P = pair<int, int>;
+
+// Validate that binary_transform_result aliases in_in_out_result
+STATIC_ASSERT(same_as<ranges::binary_transform_result<int, P, double>, ranges::in_in_out_result<int, P, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<false>{}, borrowed<false>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<ranges::dangling, ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<false>{}, borrowed<true>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<ranges::dangling, int*, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<true>{}, borrowed<false>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<int*, ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::transform(borrowed<true>{}, borrowed<true>{}, static_cast<int*>(nullptr), plus{})),
+        ranges::binary_transform_result<int*, int*, int*>>);
+
+struct instantiator {
+    static constexpr P input1[3]     = {{1, 99}, {4, 98}, {5, 97}};
+    static constexpr P input2[3]     = {{99, 6}, {98, 7}, {97, 8}};
+    static constexpr int expected[3] = {7, 11, 13};
+
+    static constexpr P shortInput1[2]     = {{1, 99}, {4, 98}};
+    static constexpr P shortInput2[2]     = {{99, 6}, {98, 7}};
+    static constexpr int shortExpected[2] = {7, 11};
+
+    template <ranges::input_range Read1, ranges::input_range Read2, weakly_incrementable Write>
+    static constexpr void call() {
+        using ranges::transform, ranges::binary_transform_result, ranges::iterator_t;
+        { // Validate iterator + sentinel overload first range shorter

```

Comment:
Comment nitpick: These are hard to read without punctuation (they don't have to be full sentences). I think that adding a comma would make "Validate blah overload, blah range shorter" readable. Occurs below.

Reply:
Added commata